pliers/extractors/vector.py

```diff
@@ -0,0 +1,67 @@
+'''
+Extractors that operate primarily or exclusively on Vector stimuli.
+'''
+from pliers.stimuli.vector import VectorStim
+from pliers.extractors.base import Extractor, ExtractorResult
+from pliers.utils import listify
+import scipy
+import numpy as np
+import pandas as pd
+from collections.abc import Iterable
+import logging
+
+class VectorMetricExtractor(Extractor):
+    ''' Extracts summary metrics from 1D-array using numpy, scipy or custom 
+        functions
+    Args:
+        functions (str, functions or list): function or list of numpy, scipy, 
+        or custom functions to be applied to 1-dimensional numpy arrays.
+            Functions can be passed directly (e.g. passing scipy.stats.entropy) if 
+            the package/module they belong to has been imported, or as strings 
+            (e.g. 'np.mean'). Custom functions returning integers or iterables 
+            can also be passed, either directly, or as strings (evaluated via 
+            eval() method).
+        var_names (optional): list of custom alias names for each metric
+        kwargs: named arguments for function call
+    ''' 
+
+    _input_type = VectorStim
+    _log_attributes = ('functions',)
+
+    def __init__(self, functions=['np.mean', 'np.std', 'np.max', 'np.min'], 
+                 var_names=None, **kwargs):
+        functions = listify(functions)
+        if var_names is not None:
+            if len(var_names) != len(functions):
+                raise ValueError('Length or var_names must match number of '
+                                 'functions')
+        for idx, f in enumerate(functions):
+            if type(f) == str:
+                f = f.replace('numpy', 'np')
```

Comment:
It's simply because numpy is already imported as np here, so forcing to to np would avoid having to reimport with full package name specification.

Reply:
Yes, that made sense to me. My point was that more generally `np` is not an "official" shortname for numpy, just a convention. So it might be safer and more genral to just generalize this to any package the user may have installed, and require the complete package name, etc..

Reply:
okay, so that implies we would try to import whatever package the function comes from? Makes sense, if it is not a problem to potentially reimport numpy here without shortname.

Reply:
read now you comment below, which makes things clear.